File: frontend/app/src/types/procesoProgramado.ts
import { TipoDeProceso } from "./tipoDeProceso";

export type ProcesoProgramado = {
  id: number;
  fechaARealizar: Date;
  frecuencia: number;
  cantidad: number;
  diaInicio: number;
  completado: Boolean;
  proceso: TipoDeProceso;
};

File: backend/src/main/java/unpsjb/labprog/backend/model/ProcesoProgramado.java
package unpsjb.labprog.backend.model;

import java.time.LocalDate;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.OneToOne;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Getter
@Setter
@NoArgsConstructor
public class ProcesoProgramado {

  @Id
  @GeneratedValue(strategy = GenerationType.AUTO)
  private Long id;

  private LocalDate fechaARealizar;

  private int frecuencia;

  private int diaInicio;

  private int cantidad;

  private Boolean completado;

  private String proceso;
  
}